Testscripts/Linux/validate-da.sh

```diff
@@ -0,0 +1,251 @@
+#!/usr/bin/env bash
+#######################################################################
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache License.
+#
+# validate-da.sh
+# Description:
+#    Validate the installation and uninstallation of the Dependency Agent.
+#    Validate the enable and disable of the Dependency Agent.
+#######################################################################
+
+set -e
+set -x
+
+# Source utils.sh
+. utils.sh || {
+    echo "ERROR: unable to source utils.sh!"
+    echo "TestAborted" > state.txt
+    exit 0
+}
+
+# Get distro information
+GetDistro
+
+readonly dir_list=(/opt/microsoft/dependency-agent /var/opt/microsoft/dependency-agent /etc/opt/microsoft/dependency-agent /lib/microsoft-dependency-agent /etc/init.d/microsoft-dependency-agent)
+
+readonly da_pid_file="/etc/opt/microsoft/dependency-agent/config/DA_PID"
+readonly uninstaller="/opt/microsoft/dependency-agent/uninstall"
+
+function test_case_cleanup(){
+    [ -f "$uninstaller" ] && eval $uninstaller
+
+    rm -rf "${dir_list[@]}"
+
+    # This can be a NOP if the test exits before calling generate_network_activity
+    kill $network_activity_generator_pid
+
+    LogMsg "Finished cleaning up test cases"
+}
+
+# Calls the cleanup function when interupted or exited
+trap test_case_cleanup SIGINT EXIT SIGHUP SIGQUIT SIGTERM
+
+function fail_test() {
+    LogErr "$*"
+    SetTestStateFailed
+    exit 0
+}
+
+function verify_directory_exists() {
+    failed_assertions=false
+    for dir in "${dir_list[@]}"
+    do
+        if [ -d "$dir" ]; then
+            LogErr "Directory $dir exists."
+            failed_assertions=true
+        fi
+    done
+
+    if $failed_assertions; then
+        LogErr "Some/All directories exists"
+        echo "$failed_assertions"
+    fi
+}
+
+function verify_distro() {
+    case $DISTRO in
+        redhat* | centos* | suse* | debian* | ubuntu*)
+            LogMsg "Supported Distro family: $DISTRO"
+            ;;
+        *)
+            fail_test "Unsupported Distro family: $DISTRO"
+            ;;
+    esac
+}
+
+function verify_expected_file() {
+    if [ ! -f "$1" ]; then
+        fail_test "${1} doesn't exist."
+    fi
+}
+
+function verify_da_pid() {
+    [ ! -f $da_pid_file ] && fail_test "$da_pid_file does not exist."
+
+    if x=$(sed -n '/^[1-9][0-9]*$/p;q' $da_pid_file); then
+        if strings /proc/$x/cmdline | fgrep -q  -e "microsoft-dependency-agent-manager"; then
+            LogMsg "PID matched"
+        else
+            fail_test "PID not matched"
+        fi
+    else
+        fail_test "PID not found in DA_PID"
+    fi
+}
+
+function check_prereqs() {
+    LogMsg "Checking for preconditions"
+
+    if [[ $EUID -ne 0 ]]; then
+        fail_test "This script must be run as root"
```

Comment:
Shouldn't this be test aborted, as it is below, versus test failed?

Reply:
Done